### ISSUE TITLE
How do I prevent this from disabling "Set Time Automatically" in Windows 11

### ISSUE THREAD

**Issue by jack-cole** [hellzerg/optimizer#496]
Whenever I restart my PC, this option is disabled. How do I prevent this from disabling it? It causes my clock to go way out of sync after a while.

![image](https://github.com/hellzerg/optimizer/assets/5460372/beb9586c-ce63-45e9-9649-de8084b1cde8)

**Comment by hellzerg** [comment]
@jack-cole Are you sure which option causes this strange behavior?

**Comment by ProphetxQc** [comment]
same problem here , dont know what setting do this...

**Comment by hellzerg** [comment]
@jack-cole @ProphetxQc 
Maybe Enable UTC time option is the problem here?

**Comment by jack-cole** [comment]
> @jack-cole @ProphetxQc Maybe Enable UTC time option is the problem here?
This is disabled in my settings.

![image](https://github.com/user-attachments/assets/5a0e89ba-aba2-4603-a5dd-eb7537b0a9e4)

**Comment by hellzerg** [comment]
@jack-cole 

You need to enable Location access in Settings > Privacy

**Comment by jack-cole** [comment]
@hellzerg

> You need to enable Location access in Windows Settings > Privacy

That allows **Set time zone automatically** to work. Even when allowing it, it will still have **Set time automatically** as disabled.

![image](https://github.com/user-attachments/assets/a6e4ea94-4f45-46aa-99b9-65b2d9f6d014)

**Comment by bkh555** [comment]
I can confirm that every troubleshooting step online fails to resolve this program borking your time function. I had to do a full repair re-installation of windows to regain proper time syncing function. There should really be some large warnings plastered across the README regarding this.

**Comment by rawhuul** [comment]
Hi, there. I am facing the same issue but able to fix it by registering `w32tm` as service and starting it.

```powershell
sudo w32tm.exe /register
sudo net start w32time
```

Additionally, if you're having issue with `Set Timezone Automatically` turned off
Enable it through registery
```powershell
sudo Set-ItemProperty -Path HKLM:\SYSTEM\CurrentControlSet\Services\tzautoupdate\ -Name Start -Value 3
```

![image](https://github.com/user-attachments/assets/c0286f26-ff95-4a6a-8550-cc731b08b0ea)

You can set your timezone either via UI or `tzutil` from powershell.

The above solved problem for me, but I am unable to find why it's been caused. I looked into the code, and it seems like `w32tm` service get disabled not directly but as a dependency of some other service.

**Comment by rawhuul** [comment]
I have created script which outputs service with its dependencies and dependents which is as follows:

```powershell
$serviceList = @()

Get-Service -CN . | ForEach-Object {
  $serviceInfo = [PSCustomObject]@{
    ServiceName       = $_.Name
    DependentServices = @()
    RequiredServices  = @()
  }

  if ($_.DependentServices) {
    $serviceInfo.DependentServices = $_.DependentServices | ForEach-Object { $_.Name }
  }

  if ($_.RequiredServices) {
    $serviceInfo.RequiredServices = $_.RequiredServices | ForEach-Object { $_.Name }
  }

  $serviceList += $serviceInfo
}

$serviceList | ConvertTo-Json -Depth 5
```

But I am not able to identify which service has dependency for `w32tm`, and it might be due to as I have enabled the service myself. However, it can be used on a clean install to identify which service has dependency or dependent on `w32tm`

**Comment by hellzerg** [comment]
@rawhuul 

The app never disables this specific service. Maybe some other similar software got this service disabled?
I don't know but this is really strange

**Comment by rawhuul** [comment]
Yup @hellzerg, I am saying the same thing. Maybe the app disable some service which has `w32tm` as dependent.

**Comment by Squtzy** [comment]
Any news on this issue? This still happens, I usually use this and ChrisTitus WinUtil on all my systems when reinstalling any system. Have made sure that it is not the WinUtil software causing this, it is always after using Optimizer that this setting gets disabled. Despite turning it back on it keeps disabling too.

**Comment by karambazielona** [comment]
The same. This app is creating a problem.
Nothing helps. 
Hi developper ? How to solve it?
![Image](https://github.com/user-attachments/assets/f9dc1f4a-0307-47cb-b5e8-352a97e570c6)

![Image](https://github.com/user-attachments/assets/62494488-9651-4762-9688-7fa85056fb7c)

**Comment by Guguroto** [comment]
I have the same problem, i just dont want to format the notebook because it just remove the "Set Time Automatically" every time i reboot the notebook. I even tried open using the sudo command that rawhuul posted, but my sudo open then closes. There is anything i can do to restore the configuration without formating?
